vllm/compilation/backends.py

```diff
@@ -164,80 +166,212 @@ def wrap_inductor(graph, example_inputs, additional_inductor_config):
 
     if additional_inductor_config is not None:
         current_config.update(additional_inductor_config)
-    if current_config['post_grad_custom_post_pass'] is not None:
-        logger.warning(
-            "post_grad_custom_post_pass is already set in the config. "
-            "Overwriting it with the fix_functionalization")
-    current_config['post_grad_custom_post_pass'] = fix_functionalization
-    return compile_fx(graph, example_inputs, config_patches=current_config)
 
+    # inductor can inplace modify the graph, so we need to copy it
+    # see https://github.com/pytorch/pytorch/issues/138980
+    graph = copy.deepcopy(graph)
+    return compile_fx(graph, example_inputs, config_patches=current_config)
 
-def vllm_backend(
-        graph,
-        example_inputs,
-        additional_inductor_config: Optional[Dict] = None) -> Callable:
 
-    context = get_compile_context()
-    context = copy.deepcopy(context) if context is not None else []
-    sizes_to_specialize: List[int] = context
+def vllm_backend(graph, example_inputs) -> Callable:
+
+    # config is read when we first compile the graph
+    # (i.e. when this backend is first called)
+    compilation_configs = CompilationConfig.default_config()
+
+    from vllm.plugins import get_non_cudagraph_ops
+    non_cudagraph_ops = get_non_cudagraph_ops()
+
+    from torch._dynamo.utils import lazy_format_graph_code
+
+    # split graph by non_cudagraph_ops
+    subgraph_id = 0
+    node_to_subgraph_id = {}
+    attention_graphs = []
+    for node in graph.graph.nodes:
```

Comment:
Nice idea to split the graph manually. Should avoid any unexpected behavior due to torch.compile internals.

Reply:
I will leave it as a TODO and followup.